src/Nethermind/Nethermind.Network.Test/ForkInfoTests.cs

```diff
@@ -30,128 +31,139 @@ namespace Nethermind.Network.Test
     [TestFixture]
     public class ForkInfoTests
     {
-        [TestCase(0, "0xfc64ec04", 1_150_000, "Unsynced")]
-        [TestCase(1_149_999, "0xfc64ec04", 1_150_000, "Last Frontier block")]
-        [TestCase(1_150_000, "0x97c2c34c", 1_920_000, "First Homestead block")]
-        [TestCase(1_919_999, "0x97c2c34c", 1_920_000, "Last Homestead block")]
-        [TestCase(1_920_000, "0x91d1f948", 2_463_000, "First DAO block")]
-        [TestCase(2_462_999, "0x91d1f948", 2_463_000, "Last DAO block")]
-        [TestCase(2_463_000, "0x7a64da13", 2_675_000, "First Tangerine block")]
-        [TestCase(2_674_999, "0x7a64da13", 2_675_000, "Last Tangerine block")]
-        [TestCase(2_675_000, "0x3edd5b10", 4_370_000, "First Spurious block")]
-        [TestCase(4_369_999, "0x3edd5b10", 4_370_000, "Last Spurious block")]
-        [TestCase(4_370_000, "0xa00bc324", 7_280_000, "First Byzantium block")]
-        [TestCase(7_279_999, "0xa00bc324", 7_280_000, "Last Byzantium block")]
-        [TestCase(7_280_000, "0x668db0af", 9_069_000, "First Constantinople block")]
-        [TestCase(9_068_999, "0x668db0af", 9_069_000, "Last Constantinople block")]
-        [TestCase(9_069_000, "0x879d6e30", 9_200_000, "First Istanbul block")]
-        [TestCase(9_199_999, "0x879d6e30", 9_200_000, "Last Istanbul block")]
-        [TestCase(9_200_000, "0xe029e991", 12_244_000, "Last Muir Glacier")]
-        [TestCase(12_244_000, "0x0eb440f6", 12_965_000, "First Berlin")]
-        [TestCase(12_964_999, "0x0eb440f6", 12_965_000, "Last Berlin")]
-        [TestCase(12_965_000, "0xb715077d", 13_773_000L, "First London")]
-        [TestCase(13_772_999, "0xb715077d", 13_773_000L, "Last London")]
-        [TestCase(13_773_000, "0x20c327fc", 15_050_000L, "First Arrow Glacier")]
-        [TestCase(15_049_999, "0x20c327fc", 15_050_000L, "Last Arrow Glacier")]
-        [TestCase(15_050_000, "0xf0afd0e3", 0L, "First Gray Glacier")]
-        [TestCase(20_000_000, "0xf0afd0e3", 0L, "Future Gray Glacier")]
-        public void Fork_id_and_hash_as_expected(long head, string forkHashHex, long next, string description)
+        [TestCase(0, 0ul, "0xfc64ec04", 1_150_000ul, "Unsynced")]
```

Comment:
Do we have any test cases with non zero timestamp for ForkId calculation?

Reply:
Even if we could add a test case for non zero timestamp, we do not know if it is the correct one since there is not spec for it yet

Reply:
I would wait for a spec to be published first. i am actually pushing for that

Reply:
I did add a few test cases with non zero timestamp in `Fork_id_and_hash_as_expected_with_timestamps` but no confirmation yet if it is a correct implementation with other clients.